src/Servers/Kestrel/Transport.Quic/test/WebHostTests.cs

```diff
@@ -20,20 +20,25 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.Tests;
 [Collection(nameof(NoParallelCollection))]
 public class WebHostTests : LoggedTest
 {
-    // This test isn't conditional on QuicListener.IsSupported. Instead, it verifies that HTTP/3 runs on expected platforms:
+    // This test isn't conditional on QuicListener.IsSupported. Instead, it verifies that HTTP/3 runs on expected CI platforms:
     // 1. Windows 11 or later.
     // 2. Linux with libmsquic package installed.
     [ConditionalFact]
+    [SkipNonHelix]
     [SkipOnAlpine("https://github.com/dotnet/aspnetcore/issues/46537")]
     [SkipOnMariner("https://github.com/dotnet/aspnetcore/issues/46537")]
     [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/46616", Queues = "Debian.12.Arm64.Open;")]
     [OSSkipCondition(OperatingSystems.MacOSX, SkipReason = "HTTP/3 isn't supported on MacOS.")]
     [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win11_21H2)]
-    public async Task PlatformSmoketest_HelloWorld_ClientSuccess()
+    public void HelixPlatform_QuickListenerIsSupported()
     {
-        // First check that System.Net.Quic reports that's supported.
         Assert.True(QuicListener.IsSupported, "QuicListener.IsSupported should be true.");
+    }
 
+    [ConditionalFact]
+    [MsQuicSupported]
+    public async Task PlatformSmoketest_HelloWorld_ClientSuccess()
```

Comment:
I want a basic request/response in the platform smoke test. There are many other tests that have `[MsQuicSupported]` that already make HTTP/3 requests.

Reply:
When the `HelixPlatform_QuickListenerIsSupported` test passes, `MsQuicSupported` is met, so this smoke test (`PlatformSmoketest_HelloWorld_ClientSuccess`) will run.

So the smoke test is here.

Should I add an assert for `MsQuicSupported.IsMet`?

Reply:
Ok, good point.

How about:
1. Remove this new `PlatformSmoketest_HelloWorld_ClientSuccess` method. Are tests that do this exact same thing.
2. Assert `MsQuicSupportedAttribute.IsMet` is true inside `HelixPlatform_QuickListenerIsSupported`

Reply:
#done